Sources/AppBundle/layout/refresh.swift

```diff
@@ -47,6 +47,25 @@ func refreshAndLayout(_ event: RefreshSessionEvent, screenIsDefinitelyUnlocked:
     refreshSession(event, screenIsDefinitelyUnlocked: screenIsDefinitelyUnlocked, startup: startup, body: {})
 }
 
+@MainActor private var havePendingRefresh = false
+@MainActor private var pendingRefreshScreenIsDefinitelyUnlocked = false
+
+@MainActor
+func scheduleRefreshAndLayout(_ event: RefreshSessionEvent, screenIsDefinitelyUnlocked: Bool = false) {
+    if havePendingRefresh {
+        if screenIsDefinitelyUnlocked {
+            pendingRefreshScreenIsDefinitelyUnlocked = true
+        }
+        return
+    }
+    havePendingRefresh = true
+    pendingRefreshScreenIsDefinitelyUnlocked = screenIsDefinitelyUnlocked
```

Comment:
It feels more natural to me to coalesce the latest `screenIsDefinitelyUnlocked` state into `pendingRefreshScreenIsDefinitelyUnlocked` in whatever case:

```suggestion
    pendingRefreshScreenIsDefinitelyUnlocked = screenIsDefinitelyUnlocked
    if havePendingRefresh { return }
    havePendingRefresh = true
```

This way we will run the next refresh session with the latest known `screenIsDefinitelyUnlocked`

wdyt?

Reply:
My logic for preferring true here was that the comments at https://github.com/nikitabobko/AeroSpace/blob/ae0e5770e5791a45773629fa19214e43ac7c94a2/Sources/AppBundle/tree/frozen/closedWindowsCache.swift#L111-L124 suggest that resetting the closed window cache on relayout is critical for correctness. Because of that, I was worried that losing a reset via https://github.com/nikitabobko/AeroSpace/blob/ae0e5770e5791a45773629fa19214e43ac7c94a2/Sources/AppBundle/layout/refresh.swift#L12 could potentially trigger that buggy sequence.

Reply:
Thanks, makes sense.

Merged with the following cleanup patch:
```diff
diff --git a/Sources/AppBundle/initAppBundle.swift b/Sources/AppBundle/initAppBundle.swift
index 38751b79..c64d0fc8 100644
--- a/Sources/AppBundle/initAppBundle.swift
+++ b/Sources/AppBundle/initAppBundle.swift
@@ -23,7 +23,7 @@ import Foundation
     AXUIElementSetMessagingTimeout(AXUIElementCreateSystemWide(), 1.0)
     startUnixSocketServer()
     GlobalObserver.initObserver()
-    refreshAndLayout(.startup1, screenIsDefinitelyUnlocked: false, startup: true)
+    refreshSession(.startup1, screenIsDefinitelyUnlocked: false, startup: true, body: {})
     refreshSession(.startup2, screenIsDefinitelyUnlocked: false) {
         if serverArgs.startedAtLogin {
             _ = config.afterLoginCommand.runCmdSeq(.defaultEnv, .emptyStdin)
diff --git a/Sources/AppBundle/layout/refresh.swift b/Sources/AppBundle/layout/refresh.swift
index 8ac24370..91cafb31 100644
--- a/Sources/AppBundle/layout/refresh.swift
+++ b/Sources/AppBundle/layout/refresh.swift
@@ -42,27 +42,16 @@ func refreshSession<T>(_ event: RefreshSessionEvent, screenIsDefinitelyUnlocked:
     return result
 }

-@MainActor
-func refreshAndLayout(_ event: RefreshSessionEvent, screenIsDefinitelyUnlocked: Bool, startup: Bool = false) {
-    refreshSession(event, screenIsDefinitelyUnlocked: screenIsDefinitelyUnlocked, startup: startup, body: {})
-}
-
 @MainActor private var havePendingRefresh = false
-@MainActor private var pendingRefreshScreenIsDefinitelyUnlocked = false

 @MainActor
 func scheduleRefreshAndLayout(_ event: RefreshSessionEvent, screenIsDefinitelyUnlocked: Bool = false) {
-    if havePendingRefresh {
-        if screenIsDefinitelyUnlocked {
-            pendingRefreshScreenIsDefinitelyUnlocked = true
-        }
-        return
-    }
+    if screenIsDefinitelyUnlocked { resetClosedWindowsCache() }
+    if havePendingRefresh { return }
     havePendingRefresh = true
-    pendingRefreshScreenIsDefinitelyUnlocked = screenIsDefinitelyUnlocked
     DispatchQueue.main.async { @MainActor in
         havePendingRefresh = false
-        refreshSession(event, screenIsDefinitelyUnlocked: pendingRefreshScreenIsDefinitelyUnlocked, startup: false, body: {})
+        refreshSession(event, screenIsDefinitelyUnlocked: false, startup: false, body: {})
     }
 }

```